mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -1303,6 +1303,74 @@ struct GridwiseAttentionAccelRewritePattern
     }
   }
 
+  void scaleFirstGemm(PatternRewriter &rewriter, Location loc,
+                      layout::GridCoordinates gridCoords, Value gemm0OutBuffer,
+                      RegsAsMatrixSubTiles gemm0OutViews, Value scaleInBuffer,
+                      Value scaleInput) const {
+    // Get current workitem ID.
+    auto tid = rewriter.create<WorkitemIdOp>(loc, rewriter.getIndexType());
+    rewriter.create<ThreadwiseReadIntoOp>(
+        loc, scaleInput, scaleInBuffer, gemm0OutViews.gridSubTile,
+        ValueRange{gridCoords.g_block, gridCoords.m_block, gridCoords.n_block,
+                   tid},
+        true, true);
+    rewriter.create<linalg::ElemwiseBinaryOp>(
+        loc, ValueRange{gemm0OutBuffer, scaleInBuffer},
+        ValueRange{gemm0OutBuffer},
+        ArrayRef<NamedAttribute>{
+            rewriter.getNamedAttr("fun", rewriter.getAttr<linalg::BinaryFnAttr>(
+                                             linalg::BinaryFn::mul)),
+            rewriter.getNamedAttr("cast", rewriter.getAttr<linalg::TypeFnAttr>(
+                                              linalg::TypeFn::cast_signed))});
+  }
+
+  FailureOr<TypedAttr>
+  getSplatGlobalConstant(memref::GetGlobalOp getGlobalOp) const {
+    auto *symbolTableOp =
+        getGlobalOp->getParentWithTrait<OpTrait::SymbolTable>();
+    if (!symbolTableOp)
+      return failure();
+    auto global = dyn_cast_or_null<memref::GlobalOp>(
+        SymbolTable::lookupSymbolIn(symbolTableOp, getGlobalOp.getNameAttr()));
```

Comment:
I think there's a way to look up the nearest symbol, you might want to use that?

Reply:
any pointers would be appreciated.

Reply:
`lookupNearestSymbolFrom` is a static method in `SymbolTable`

Reply:
thanks! now changed